ncm-network/src/main/perl/networkmanager.pm

```diff
@@ -0,0 +1,611 @@
+#${PMpre} NCM::Component::networkmanager${PMpost}
+
+=head1 NAME
+
+network: Configure Network Settings using NetworkManager Keyfile format. Most functions and logic is taken from network module. 
+
+=head1 DESCRIPTION
+
+The I<network> component sets the network settings through C<< /etc/sysconfig/network >>
+and the keyfile format files in C<< /etc/NetworkManager/system-connections >>.
+
+New/changed settings are first tested by retrieving the latest profile from the
+CDB server (using ccm-fetch).
+If this fails, the component reverts all settings to the previous values.
+
+During this test, a sleep value of 15 seconds is used to make sure the restarted network
+is fully restarted (routing may need some time to come up completely).
+
+Because of this, configuration changes may cause the ncm-ncd run to take longer than usual.
+
+Be aware that configuration changes can also lead to a brief network interruption.
+=cut
+
+use parent qw (NCM::Component::network);
+
+our $EC = LC::Exception::Context->new->will_store_all;
+
+our $NoActionSupported = 1;
+
+use Readonly;
+
+Readonly my $HOSTNAME_CMD => '/usr/bin/hostnamectl';
+
+Readonly my $NETWORK_PATH => '/system/network';
+Readonly my $HARDWARE_PATH => '/hardware/cards/nic';
+Readonly my $IFCFG_DIR => "/etc/NetworkManager/system-connections";
```

Comment:
reason for keeping this as $IFCFG_DIR was to see if  when it creates backup in this path, but it still doesn't. creates backup dir in network-scripts location